docs/contributor/image_mode.md

```diff
@@ -0,0 +1,308 @@
+permissions configured.
+
+### Configure CNI
+
+The MicroShift CNI driver (OVN) requires the Open vSwitch service to function
+properly. The service depends on the `openvswitch` kernel module to be available
+in the `bootc` image.
+
+Run the following commands on the host to check the `openvswitch` module presence
+and the version of `kernel-core` package used in the `bootc` image. Note that the
+kernel versions are different.
+
+```
+$ find /lib/modules/$(uname -r) -name "openvswitch*"
+/lib/modules/6.9.9-200.fc40.x86_64/kernel/net/openvswitch
+/lib/modules/6.9.9-200.fc40.x86_64/kernel/net/openvswitch/openvswitch.ko.xz
+
+$ IMAGE_NAME=microshift-4.16-bootc
+$ sudo podman inspect "${IMAGE_NAME}" | grep kernel-core
+        "created_by": "kernel-core-5.14.0-427.26.1.el9_4.x86_64"
+```
+
+When a `bootc` image is started, it uses the host kernel, which is not necessarily
+the same one used for building the image. This means that the `openvswitch` module
+cannot be loaded in the container due to the kernel version mismatch with the
+modules present in the `/lib/modules` directory.
+
+To work around this problem, it is necessary to share the host `/lib/modules`
+directory with the container when it is started. This is accomplished by providing
+the `-v /lib/modules:/lib/modules:ro` option to the `podman run` command as
+described below.
+
+### Configure CSI
+
+If the host is already configured to have a `rhel` volume group with free space,
+this configuration is inherited by the container so that it can be used by the
+MicroShift CSI driver to allocate storage.
+
+```
+$ sudo vgs
+  VG   #PV #LV #SN Attr   VSize   VFree
+  rhel   1   1   0 wz--n- <91.02g <2.02g
+```
+
+Additional configuration is required for using storage in `bootc` MicroShift
+images when the host does not have the volume group setup as required by the
+MicroShift CSI driver.
+
+Run the following commands to create a file to be used for LVM partitioning and
+set it up as a loop device.
+
+```
+VGFILE=/var/lib/microshift-lvm-storage.img
+VGSIZE=1G
+
+sudo truncate --size="${VGSIZE}" "${VGFILE}"
+sudo losetup -f "${VGFILE}"
+```
+
+Query the loop device name and create a free volume group on the device according
+to the MicroShift CSI driver requirements.
+
+```
+VGLOOP=$(losetup -j ${VGFILE} | cut -d: -f1)
+sudo vgcreate -f -y rhel "${VGLOOP}"
+```
+
+The device will now be shared with the newly created containers as described in
+the next section.
+
+> The following commands can be run to detach the loop device and delete the LVM
+> volume group file.
+> ```
+> sudo losetup -d "${VGLOOP}"
+> sudo rm -f "${VGFILE}"
+> ```
+
+### Run Container
+
+Run the following command to start the MicroShift `bootc` image in an interactive
+terminal session. Note that the host container storage and kernel modules paths
+are shared with the container.
+
+```
+IMAGE_NAME=microshift-4.16-bootc
+
+sudo podman run --rm -it --privileged \
```

Comment:
After everything described here I'd be a bit wary of running all this stuff as a privileged container...how sure are you it's not doing unexpected things in the host?

Reply:
That's an interesting question and I do not have a good answer for it, except that this is a development flow, not a production one. I will discuss with MicroShift PM on whether we'd like to add this to downstream docs. 

The production flow is running a VM and applying an image on top of it.

Reply:
After confirming with MicroShift PM, we are not going to publish how to run MicroShift containers in product docs, only in the repo contributor docs.